tests/deepspeed/test_deepspeed.py

```diff
@@ -1051,3 +1074,46 @@ def test_lr_scheduler(self):
         ]
         with patch_environment(omp_num_threads=1):
             execute_subprocess_async(cmd)
+
+    @require_cuda
+    def test_basic_dynamo_run(self):
+        self.test_file_path = self.test_scripts_folder / "test_performance.py"
+        with tempfile.TemporaryDirectory() as dirpath:
+            cmd = [
+                "accelerate",
+                "launch",
+                "--num_processes=1",
+                "--num_machines=1",
+                "--machine_rank=0",
+                "--mixed_precision=fp16",
+                "--use_deepspeed",
+                "--gradient_accumulation_steps=1",
+                "--zero_stage=3",
+                "--offload_optimizer_device=none",
+                "--offload_param_device=none",
+                "--dynamo_backend=eager",
+                self.test_file_path,
+                "--model_name_or_path=distilbert-base-uncased",
+                "--num_epochs=1",
+                f"--output_dir={dirpath}",
+            ]
+            with patch_environment(omp_num_threads=1):
+                with_dynamo = False
+                try:
+                    r = execute_subprocess_async(
+                        cmd,
+                        env={
+                            "TORCH_LOGS": "dynamo",
+                            # dynamo itself has some issue, use below to only compile `forward` for testing.
+                            # On deepspeed side, `deepspeed.util.z3_leaf_module.[un]set_z3_leaf_modules` is used for similar issue
+                            # that user want to compile/skip specific module.
+                            "TORCHDYNAMO_DEBUG_FUNCTION": "forward",
```

Comment:
Why is this required? The above comment is unclear. A detailed explanation in this PR and a slightly detailed comment would help to know what needs to be done to get DeepSpeed+Compile working.

Reply:
Dynamo still don't all python op and may cause graph break and even failure during compile. However I didn't dive into the detail of the failure and use this whitelist to tell dynamo to only compile `forward` function, which will also break a module into several functions. This is why you didn't see improvement comparing inductor and no dynamo.

Based on my experience using dynamo with large model, user should specify which module they want to compile as one, which can be done with mentioned deepspeed api with user code modified. The focus of the test in this PR is on determining if dynamo is enabled, and I did not evaluate its performance.

Reply:
In our internal scenario, dynamo can perform a ~10% speedup if compile each `LLamaDecodeLayer`.